caffe2/operators/batch_gather_ops.cc

```diff
@@ -12,24 +12,20 @@ OPERATOR_SCHEMA(BatchGather)
                                 const vector<TensorShape>& in) {
       vector<TensorShape> out(1);
       ArgumentHelper helper(def);
-
-      vector<int> output_dims;
       const auto& data_dims = GetDimsVector(in[0]);
       const auto& indices_dims = GetDimsVector(in[1]);
-      output_dims.push_back(data_dims[0]);
-      output_dims.insert(
-          output_dims.end(), indices_dims.begin(), indices_dims.end());
-      output_dims.insert(
-          output_dims.end(), data_dims.begin() + 2, data_dims.end());
 
+      vector<int> output_dims =
+          caffe2::gather_helper::calc_output_shape_vector<int>(
+              data_dims, indices_dims, 1);
       out[0] = CreateTensorShape(output_dims, TensorProto::FLOAT);
       return out;
     })
     .SetDoc(R"DOC(
 Batch gather operation, first dimension in DATA is the batch size.
 Given DATA tensor of rank r >= 2, and INDICES tensor of rank q >= 1, gather
 entries of the outer-most dimension of DATA indexed by INDICES, and concatenate
```

Comment:
it's not outer-most dimension any more (this description was wrong before)

Reply:
will update!